core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -199,6 +200,22 @@ class CachedDataFrame(df: Dataset[Row], var queryString: String,
     collectInternal().map(boundEnc.fromRow).toArray
   }
 
+  override def withNewExecutionId[T](body: => T): T = queryExecution.executedPlan match {
+    // don't create a new executionId for ExecutePlan since it has already done so
+    case _: ExecutePlan => body
+    case _ =>
```

Comment:
ExecutePlan does the actual execution in its plan itself so that one does not need to do a collect() for insert/delete/update. However, if user does do a collect (e.g. to get the number of rows updated/deleted/..), then it will show up separately. Is there a way to connect the two if second one is present or perhaps that is not required? Test with sql("insert into ...") with and without collect() and see how the two look. Maybe the main query is shown in first while second appears as dummy which is fine.